src/SpyingFileFetcher.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace FileFetcher;
+
+/**
+ * Decorator for FileFetcher objects that records file fetching calls.
+ *
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class SpyingFileFetcher implements FileFetcher {
+
+	private $fileFetcher;
+
+	private $fetchedUrls = [];
+
+	public function __construct( FileFetcher $fileFetcher ) {
+		$this->fileFetcher = $fileFetcher;
+	}
+
+	/**
+	 * @see FileFetcher::fetchFile
+	 * @throws FileFetchingException
```

Comment:
For some docu generating thing? I don't know what this does...

I care about being able to see the contract of the thing when looking at the code.